sdk/keyvault/azure-security-keyvault-certificates/src/certificate_client_operations.cpp

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#include "azure/keyvault/certificates/certificate_client_operations.hpp"
+#include "azure/keyvault/certificates/certificate_client.hpp"
+#include "azure/keyvault/certificates/certificate_client_models.hpp"
+#include "private/certificate_serializers.hpp"
+#include <thread>
+
+using namespace Azure::Security::KeyVault::Certificates;
+
+Azure::Response<KeyVaultCertificate> CreateCertificateOperation::PollUntilDoneInternal(
+    std::chrono::milliseconds period,
+    Azure::Core::Context& context)
+{
+  while (true)
+  {
+    Poll(context);
+    if (IsDone())
+    {
+      break;
+    }
+    std::this_thread::sleep_for(period);
+  }
+
+  return Azure::Response<KeyVaultCertificate>(
+      m_value, std::make_unique<Azure::Core::Http::RawResponse>(*m_rawResponse));
+}
+
+std::unique_ptr<Azure::Core::Http::RawResponse> CreateCertificateOperation::PollInternal(
+    Azure::Core::Context const& context)
+{
+  std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse;
+
+  try
+  {
+    rawResponse = m_certificateClient->GetCertificate(m_value.Name(), context).RawResponse;
+  }
+  catch (Azure::Core::RequestFailedException& error)
```

Comment:
```suggestion
  catch (Azure::Core::RequestFailedException const& error)
```
? Or move won't compile then?